src/main/scala/com/scality/clueso/CluesoConstants.scala

```diff
@@ -0,0 +1,93 @@
+package com.scality.clueso
+
+import org.apache.spark.sql.functions.col
+import org.apache.spark.sql.types._
+
+object CluesoConstants {
+  // order does matter:
+
+  val replicationInfoSchema = new StructType()
+    .add("status", StringType, true)
+    .add("content", new ArrayType(StringType, false), true)
+    .add("destination", StringType, true)
+    .add("storageClass", StringType, true)
+    .add("role", StringType, true)
+
+  val locationSchema = new StructType()
+    .add("key", StringType, false)
+    .add("size", LongType, false)
+    .add("start", LongType, false)
+    .add("dataStoreName", StringType, true)
+    .add("dataStoreETag", StringType, true)
+
+  val eventAclSchema = new StructType()
+    .add("Canned", StringType)
+    .add("FULL_CONTROL", new ArrayType(StringType, false))
+    .add("WRITE_ACP", new ArrayType(StringType, false))
+    .add("READ", new ArrayType(StringType, false))
+    .add("READ_ACP", new ArrayType(StringType, false));
+
+  val eventValueSchema = new StructType()
+    .add("userMd", MapType(StringType, StringType, true), true)
+    .add("bucket", StringType)
+    .add("incrementer", StringType)
+    .add("sourceFileName", StringType)
+    .add("key", StringType)
+    .add("acl", new StructType(eventAclSchema.fields), false)
+    .add("location", new ArrayType(locationSchema, false))
+    .add("tags", new MapType(StringType, StringType, false), true)
+    .add("replicationInfo", replicationInfoSchema)
+    .add("md-model-version", IntegerType)
+    .add("owner-display-name", StringType)
+    .add("owner-id", StringType)
+    .add("content-length", IntegerType)
+    .add("content-type", StringType)
+    .add("last-modified", StringType)
```

Comment:
I will have to check if the format is automatically CASTable in this phase – I know we're doing the cast but when reading the schema I don't know if we assume that because format matters.